src/azul/plugins/metadata/hca/transform.py

```diff
@@ -332,16 +335,17 @@ class Submitter(SubmitterBase, Enum):
         SubmitterCategory.external
     )
 
-    def __init__(self, id: str, title: str, category: SubmitterCategory):
+    def __init__(self, submitter_id: str, title: str, category: SubmitterCategory):
         super().__init__()
         slug = self.name.replace('_', ' ')
         generated_uuid = str(uuid5(self.id_namespace, slug))
-        assert id == generated_uuid, (id, generated_uuid)
-        self.id = id
+        assert submitter_id == generated_uuid, (submitter_id, generated_uuid)
+        self.id = submitter_id
         self.slug = slug
         self.title = title
         self.category = category
-        self.by_id[self.id] = self
+        self.id_by_title[title] = submitter_id
```

Comment:
```suggestion
        self.by_title[title] = self
```

is more flexible because it allows you to lookup any of the properties by title, not just id.

Reply:
Assert that there are no duplicate titles, IOW that this line actually adds an entry as opposed to overwrite an existing one.